apps/src/music/player/MusicLibrary.ts

```diff
@@ -69,6 +72,47 @@ export async function loadLibrary(libraryName: string): Promise<MusicLibrary> {
       );
     }
 
+    // Second, retrieve any partial library files.  These contain
+    // additional content that will be merged into the library.
+    const partialsPromises: Promise<GetResponse<LibraryJson>>[] | undefined =
+      libraryJson.partials?.map(partial => {
+        return HttpClient.fetchJson<LibraryJson>(
+          getBaseAssetUrl() +
+            'v2/partials/' +
+            partial +
+            '.json' +
+            (requestVersion ? `?version=${requestVersion}` : ''),
+          {},
+          LibraryValidator
+        );
+      });
+
+    let partialsResponses: PromiseSettledResult<GetResponse<LibraryJson>>[] =
+      [];
+    if (partialsPromises) {
+      partialsResponses = await Promise.allSettled(partialsPromises);
```

Comment:
minor: why not use `Promise.all`? Is it because we don't want to fail if an individual partial fails to load?

Reply:
Good question.  `Promise.all` is probably the right thing.  I was replicating [this](https://github.com/code-dot-org/code-dot-org/pull/60460/files#diff-ec37ec497bc210dfd0dbfd593486019576a2579baac851b9f55866607218d5e2R57-R59), but @bencodeorg, perhaps that was done because the translation file is not always requested?